src/index.js

```diff
@@ -19,329 +19,331 @@ const DrumPlayer = require('./DrumPlayer');
-        Tone.Master
-    );
-
-    // global tempo in bpm (beats per minute)
-    this.currentTempo = 60;
-
-    // instrument player for play note blocks
-    this.instrumentPlayer = new InstrumentPlayer(this.input);
-    this.numInstruments = this.instrumentPlayer.instrumentNames.length;
-
-    // drum player for play drum blocks
-    this.drumPlayer = new DrumPlayer(this.input);
-    this.numDrums = this.drumPlayer.drumSounds.length;
-
-    // a map of md5s to audio buffers, holding sounds for all sprites
-    this.audioBuffers = {};
-
-    // microphone, for measuring loudness, with a level meter analyzer
-    this.mic = null;
-    this.micMeter = null;
-};
+class AudioEngine {
+    constructor () {
+        // create the global audio effects
+        this.roboticEffect = new RoboticEffect();
+        this.fuzzEffect = new FuzzEffect();
+        this.echoEffect = new EchoEffect();
+        this.reverbEffect = new ReverbEffect();
+
+        // chain the global effects to the output
+        this.input = new Tone.Gain();
+        this.input.chain(
+            this.roboticEffect, this.fuzzEffect, this.echoEffect, this.reverbEffect,
+            Tone.Master
+        );
+
+        // global tempo in bpm (beats per minute)
+        this.currentTempo = 60;
+
+        // instrument player for play note blocks
+        this.instrumentPlayer = new InstrumentPlayer(this.input);
+        this.numInstruments = this.instrumentPlayer.instrumentNames.length;
+
+        // drum player for play drum blocks
+        this.drumPlayer = new DrumPlayer(this.input);
+        this.numDrums = this.drumPlayer.drumSounds.length;
+
+        // a map of md5s to audio buffers, holding sounds for all sprites
+        this.audioBuffers = {};
+
+        // microphone, for measuring loudness, with a level meter analyzer
+        this.mic = null;
+        this.micMeter = null;
+    }
 
-/**
- * Decode a sound, decompressing it into audio samples.
- * Store a reference to it the sound in the audioBuffers dictionary, indexed by md5
- * @param  {object} sound - an object containing audio data and metadata for a sound
- * @property {Buffer} data - sound data loaded from scratch-storage.
- * @property {string} format - format type, either empty or adpcm.
- * @property {string} md5 - the MD5 and extension of the sound.
- * @returns {?Promise} - a promise which will resolve after the audio buffer is stored, or null on error.
- */
-AudioEngine.prototype.decodeSound = function (sound) {
-
-    let loaderPromise = null;
-
-    switch (sound.format) {
-    case '':
-        loaderPromise = Tone.context.decodeAudioData(sound.data.buffer);
-        break;
-    case 'adpcm':
-        loaderPromise = (new ADPCMSoundDecoder()).decode(sound.data.buffer);
-        break;
-    default:
-        return log.warn('unknown sound format', sound.format);
+    /**
+     * Names of the audio effects.
+     * @enum {string}
+     */
+    static get EFFECT_NAMES () {
```

Comment:
This doesn't seem to work (e.g. click the "set effect" block to get an error because this.audioEngine.EFFECT_NAMES is undefined). 

I tried making EFFECT_NAMES a property of the audioEngine by adding it to the constructor, and that works fine.